### ISSUE TITLE
[Bug] With monocle 0.1.7-alpha3, not all the blocks can be illuminated even if they are exposed to sun

### ISSUE THREAD

**Issue by chemlzh** [ferriarnus/Monocle#3]
Mod added: embeddium 1.0.11+iris 1.8.0-beta5+monocle 0.1.7-alpha3.

Shader packs tested: Complementary (Reimaged/Unbound), Aurora's Shaders, Makeup Ultrafast

Hardware: no independent graphic card used, only Intel UHD Graphics 750 working.

Problem: not all the blocks can be illuminated even if they are exposed to sun (Complementary Reimaged for example).

![image](https://github.com/user-attachments/assets/90dd927a-543e-45a9-b499-3e59a925cce2)

**Comment by ferriarnus** [comment]
Hmm, are all grass blocks supposed to look the same?

**Comment by chemlzh** [comment]
> Hmm, are all grass blocks supposed to look the same?

Yes, since I have generated a flat world with grass blocks. And below is Fabric 1.21.1+Sodium 0.6.0-beta2+iris 1.8.0-alpha4 (same shaderpack)

![image](https://github.com/user-attachments/assets/1b74115e-be5b-4fd3-aa89-1935c17d337e)

**Comment by ferriarnus** [comment]
Ok, thank you for letting me know. Can you also send a log from when you tested this? You are using windows with intergrated intell graphics?

**Comment by chemlzh** [comment]
> Ok, thank you for letting me know. Can you also send a log from when you tested this? You are using windows with intergrated intell graphics?

[latest.log](https://github.com/user-attachments/files/17492654/latest.log)
[debug.log](https://github.com/user-attachments/files/17492659/debug.log)

Yes, I'm running on Windows with Intel graphics.

**Comment by chemlzh** [comment]
Still not work with updated monocle 0.1.7-alpha3

![image](https://github.com/user-attachments/assets/8ce6991f-f4cd-41d3-a158-acf171327423)

The debug and latest logs are attached:

[debug.log](https://github.com/user-attachments/files/17499571/debug.log)

[latest.log](https://github.com/user-attachments/files/17499573/latest.log)

Here's the comparison (Fabric):

![image](https://github.com/user-attachments/assets/844ac5da-2c27-4cfd-aa33-04de74cf0115)

**Comment by chemlzh** [comment]
And here is sodium 0.6.0-beta2+iris 1.8.0-beta5 on Neoforge 1.21.1

![image](https://github.com/user-attachments/assets/aeb2eee5-f4bc-47a1-827b-953763989204)

**Comment by SodiumZH** [comment]
This seems not to happen with NVIDIA GPU. Maybe it's related to compatibility with Intel integrated GPU?

**Comment by chemlzh** [comment]
Now with embeddium 1.0.11+iris 1.8.1+monocle 0.1.8, the shader finally renders correctly. (I use renderPearl for testing) Thanks very much!!!!!

![Final result in monocle 0.1.8](https://github.com/user-attachments/assets/8d2cf66b-8e5d-4db8-a984-e74b1c1056a7)

![Final result in monocle 0.1.8 at night](https://github.com/user-attachments/assets/ea89563b-76f2-4a42-9081-b38485a48b9f)
